### DISCUSSION TITLE
Vertical space tool, default image size

### DISCUSSION THREAD

**Discussion by Doublonmousse** [flxzt/rnote#721] [Ideas]
I created a fork and tried to implement new features, but before opening a PR, I would like to discuss what I've done. Also, this is more of a "personal fix that works ok" than a properly coded addition at this stage :

- Vertical Space tool that aligns with grids :
![vtool](https://github.com/flxzt/rnote/assets/115779707/c417e579-428e-48c7-a33e-79a5b8b3919e)

- Set the default size of an image imported to be a fraction of the page width : For now, I activated this on every format, but I feel like such an option is good for fixed and continuous vertical modes (where resizing an image that is too large is cumbersome), and optional for the other two (meaning we can discuss if this is necessary or not). 
Also, for now the size is set to 70 % of the page width, and hard coded, but putting it in the settings pane should be a better implementation. Once again, the code could be improved.

Should I open a PR ?

**Comment by flxzt** [comment]
Aligning/snapping to the grid is a useful feature, but we should think about adding this as an option globally that would work across different tools / shapes.

I am not sure about the image size. I want to avoid an import dialog / additional settings for images, at the same time I can see that automatically scaled images to the current view or page dimensions might be useful in some cases. Maybe there can be improved behaviour that would work best for most users, for example:  `width = image_width.min(page_width)`.

**Reply by Doublonmousse** [reply]
Yes, I agree that a global "align to grid" option that enables this option across different tools (and this behavior of the vertical space toolbar) would make sense.

For the image size, I agree that ```width = image_width.min(page_width)``` would be better.  
At least for the vertical modes. 
I don't know if it should be this way for the semi infinite and infinite mode as they can be used without taking into account format borders.

Another idea would be to scale the image based on the zoom level (changing ```page_width``` by the width of the page visible at the current level of zoom, denoted with some ```zoom_width```). This way the image can fit on the vertical modes and also be of reasonable size with respect to the rest of the page for infinite modes.
